core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -148,7 +148,13 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** Cleanup local dirs and stop shuffle sender. */
   private[spark] def stop() {
     // Remove the shutdown hook.  It causes memory leaks if we leave it around.
-    Runtime.getRuntime.removeShutdownHook(shutdownHook)
+    if(shutdownHook != null) {
+      try {
+        Runtime.getRuntime.removeShutdownHook(shutdownHook)
+      } catch {
+        case e: Exception => None
```

Comment:
Narrow this to `IllegalStateException`, to only squash this particular case, hopefully? I think leaving the `null` check in is OK even if I'm not sure it can happen since the variable is assigned to non-`null` in its constructor. Nit: put a space after `if`. I'll pause a beat for others to weigh in today.